challenges-46-50.yml

```diff
@@ -0,0 +1,219 @@
+              <div class="alert alert-success">
+                  {{ flash }}
+              </div>
+          {% endfor %}
+          </div>
+          ```
+        c: |
+          ```twig
+          <div id="flash-container">
+              <meta name="data-turbo-cache" content="false">
+              {% for flash in app.session.flashBag.get('success') %}
+                  <div class="alert alert-success">
+                      {{ flash }}
+                  </div>
+              {% endfor %}
+          </div>
+          ```
+        d: |
+          Both A and C are correct.
+      correct: a
+
+mercure:
+  question: |
+    What is Mercure?
+  explanation: |
+    The Mercure service is the go to application when building interactive web
+    apps such as chatrooms and the likes.
+    
+    Oh, and the Planet is called `Mercury`! :p
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          Mercure is the first planet of our Solar System!
+        b: |
+          Mercure is a service that runs in a host that allows messages to be
+          pushed to web clients without having to perform an http request!
+        c: |
+          Mercure is a browser extension for Firefox and Chrome that allows
+          us to send push notifications to them!
+        d: | 
+          Both A and B are correct!
+      correct: b
+
+listen_publish:
+  question: |
+    My chat application is finally ready to become interactive! I've just
+    installed the `symfony/ux-turbo-mercure` package and set up Mercure in
+    my server and I'm ready to receive messages from other people!
+    
+    My current template is looking like this:
+    
+    ```twig
+    <turbo-stream action="update" target="chat-messages">
+        <template>
+            {{ include('chat/_messages.html.twig') }}
+        </template>
+    </turbo-stream>
+
+    <turbo-stream action="replace" target="chat-user-list">
+        <template>
+            {{ include('chat/_user-list.html.twig') }}
+        </template>
+    </turbo-stream>
+    ```
+    
+    And I now need to listen to a `new-messages` topic in Mercure in order to
+    receive them. How is the way to do it?
+  explanation: |
+    That twig function is all you need to make your page listen to that
+    Mercure connection. It's not a *visual* component (e.g. It will not be 
+    visible in any way in our page) but a functional one!
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```twig
+          <turbo-stream data-mercure-topic="new-messages" action="update" target="chat-messages>
+              <template>
+                  {{ include('chat/_messages.html.twig') }}
+              </template>
+          </turbo-stream>
+          ```
+        b: |
+          ```twig
+          <turbo-frame data-mercure-topic="new-messages">
+              {{ include('chat/_messages.html.twig') }}
+          <turbo-frame>
+          ```
+        c: |
+          Both A and D are correct.
```

Comment:
It looks weird to not have this option last?

Also, does/will our new system randomize the order of the answers? If so, I'm realizing that answers like "Both A and D are correct" will break because A might be C and D might be B!

Reply:
I think answers such as these won't be randomized at all. My proposal to invent a syntax (for DB challenges) that we can replace with the actual letter of the answer wasn't accepted. May be I can move it to D.